src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -60,15 +68,71 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
         this.paramSubscription.unsubscribe();
     }
 
-    public groupExercises(selectedOrder: number): void {
+    /**
+     * Return the total number of exercises for the logged in user
+     */
+    getNrOfExercises(): number {
+        return [...this.exerciseCountMap.values()].reduce((a, b) => a + b, 0);
+    }
+
+    /**
+     * Reorders all displayed exercises
+     * @param selectedOrder The order in which the exercises should be displayed
+     */
+    orderUpdate(selectedOrder: number) {
+        this.order = selectedOrder;
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Filters all displayed exercises by applying the selected filters
+     * @param filters The filters which should be applied
+     * @param active Should the filters be active or inactive
+     */
+    filterUpdate(filters: number[], active: boolean) {
+        if (active) {
+            filters.forEach(filter => this.filters.add(filter));
+        } else {
+            filters.forEach(filter => this.filters.delete(filter));
+        }
+
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Checks if the given exercise still needs work, i.e. is not graded with 100%, or wasn't even started, yet.
+     * @param exercise The exercise which should get checked
+     */
+    private needsWork(exercise: Exercise): boolean {
+        console.log(this.user);
```

Comment:
Please remove console.logs, we don't have it in the linter because we sometimes do it (we could talk about adding it as a rule in my opinion).

Reply:
removed